### PR TITLE
io.github.yuxshao.ptcollab

### PR DESCRIPTION
**Pull request by germe-deb**
### Please confirm your submission meets all the criteria

Please describe your application briefly below.

pxtone collab (ptcollab) is a multi user sample-based music editor. It's based off the music engine of pxtone, a music editor developed by Pixel, the creator of Cave Story and Kero Blaster.
ptcollab can be used to create chiptune music, but because you can import any sample, it can also be used to make regular music, like classical, or pop or more!
You can connect with your friends over internet and edit music together!

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* (1)
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://discord.com/channels/237299857579180033/237299857579180033/1212546882832826419** (2)
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

(1): permissions of the app:
```
--device=dri           access hardware acceleration
--share=network        allow you to connect to other player's host, or host a session
--socket=wayland       allow the app to display a window in wayland
--socket=pulseaudio    allow the app to sound
--socket=fallback-x11  allow the app to display a window in xorg
--filesystem=home      allow the app open and save files to the hard drive
```

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: io.github.yuxshao.ptcollab.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    buildsystem: qmake
+    sources:
+      - type: git
+        url: https://github.com/yuxshao/ptcollab.git
+        tag: "v0.6.4.8"
+        commit: "5feb27decb9e0600d1defdaf9dc1424c6fdcb7ad"
+  - name: share
+    buildsystem: simple
+    sources:
+      - type: dir
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:17]: 1. don't use `dir` source
2. only the SVG icon is necessary. remove all the others
3. no need for a separate module

File: io.github.yuxshao.ptcollab.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        path: share
+    build-commands:
+      - rm /app/share/applications -rf
+      - rm /app/share/icons/hicolor/*/*/*
+      - cp -rf ../share /app/
+finish-args:
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:23]: this section should be found above `modules`.

File: io.github.yuxshao.ptcollab.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        tag: "v0.6.4.8"
+        commit: "5feb27decb9e0600d1defdaf9dc1424c6fdcb7ad"
+  - name: share
+    buildsystem: simple
+    sources:
+      - type: dir
+        path: share
+    build-commands:
+      - rm /app/share/applications -rf
+      - rm /app/share/icons/hicolor/*/*/*
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:21]: why do you remove this?

File: share/metainfo/io.github.yuxshao.ptcollab.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <ul>
+      <li>Collaborative editing: host a server and have others connect and work with you!</li>
+	  <li>Session recording: save a frame-by-frame recording of a session and play it back later!</li>
+      <li>MIDI input: Record notes either live while the song is playing, or using keyboard controls to step through time.</li>
+    </ul>
+  </description>
+  <launchable type="desktop-id">io.github.yuxshao.ptcollab.desktop</launchable>
+  <content_rating type="oars-1.1">
+    <content_attribute id="social-chat">mild</content_attribute>
+  </content_rating>
```
Comment (hfiguiere) [inline share/metainfo/io.github.yuxshao.ptcollab.metainfo.xml:38]: missing `releases` and `screenshots`

File: io.github.yuxshao.ptcollab.yml (hunk 1 above)
Comment (hfiguiere) [reply io.github.yuxshao.ptcollab.yml:17]: actually the icons are upstream they don't need to be here at all.

File: share/metainfo/io.github.yuxshao.ptcollab.metainfo.xml (hunk 4 above)
Comment (germe-deb) [reply share/metainfo/io.github.yuxshao.ptcollab.metainfo.xml:47]: screenshots added!
I'm trying to figure out how releases work

File: io.github.yuxshao.ptcollab.yml (hunk 2 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml]: Ok, done

File: io.github.yuxshao.ptcollab.yml (hunk 1 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:25]: Ok

File: share/icons/hicolor/16x16/apps/io.github.yuxshao.ptcollab.png
Comment (hfiguiere) [inline share/icons/hicolor/16x16/apps/io.github.yuxshao.ptcollab.png:1]: remove all these icons, use upstream

File: share/applications/io.github.yuxshao.ptcollab.desktop
Comment (hfiguiere) [inline share/applications/io.github.yuxshao.ptcollab.desktop:1]: remove this, use upstream

File: io.github.yuxshao.ptcollab.yml (hunk 3 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:29]: Ohh i see

File: io.github.yuxshao.ptcollab.yml (hunk 1 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:25]: I have a problem with this, the original project doesn't has a metainfo file, I need somewhere to store it

Comment (hfiguiere) [reply io.github.yuxshao.ptcollab.yml:25]: I didn't mention that one, so what's the problem?

Comment (hfiguiere) [reply io.github.yuxshao.ptcollab.yml:25]: albeit it should be submitted for inclusion upstream for a future release (including installing it), as it's not flatpak specific.

File: share/icons/hicolor/16x16/apps/io.github.yuxshao.ptcollab.png
Comment (germe-deb) [reply share/icons/hicolor/16x16/apps/io.github.yuxshao.ptcollab.png:1]: this is complicating me a lot, isn't there any other solution? the share dir worked fine

**Comment by germe-deb** [comment]
at the moment the program no longer compiles, I might backtrack to the moment I had the extra module for metadata, as at that moment, ptcollab not only compiled, but also launched and worked

File: io.github.yuxshao.ptcollab.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+id: io.github.yuxshao.ptcollab
+runtime: org.kde.Platform
+runtime-version: '5.15-23.08'
+sdk: org.kde.Sdk
+command: ptcollab
+rename-icon: ptcollab.png
+# rename-icon: ptcollab.svg
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:7]:
```suggestion
rename-icon: ptcollab
```

File: io.github.yuxshao.ptcollab.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+# rename-desktop-file: ptcollab.desktop
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:8]:
```suggestion
rename-icon: ptcollab
rename-desktop-file: ptcollab.desktop
```

File: io.github.yuxshao.ptcollab.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  <id>io.github.yuxshao.ptcollab</id>
+  
+  <name>pxtone Collab</name>
+  <summary>Sample-based music editor where you can collaborate with friends!</summary>
+  
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
+
+  <screenshots>
+    <screenshot type="default">https://cdn.discordapp.com/attachments/1033076370420535398/1212564674097447032/Captura_de_pantalla_de_2024-02-28_21-51-40.png?ex=65f24bda&is=65dfd6da&hm=bc8aeeebc65e67e3c64b7d5a8fee59d103810b90ed6a2b561216fc87ede10233&</screenshot>
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.metainfo.xml:12]: url have to be inside `<image>`

Don't use discord as your storage. It's unreliable at best.

File: io.github.yuxshao.ptcollab.metainfo.xml
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.metainfo.xml:1]: `developer_name` and `releases` still missing. they are mandatory

File: io.github.yuxshao.ptcollab.metainfo.xml (hunk 7 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:12]: Oh ok! what service do I use then? Can I use protondrive?

File: io.github.yuxshao.ptcollab.metainfo.xml
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: Now the xml has releases

File: io.github.yuxshao.ptcollab.metainfo.xml (hunk 7 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml]: this git repo? or another one?

File: io.github.yuxshao.ptcollab.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    buildsystem: qmake
+    build-commands:
+    sources:
+      - type: git
+        url: https://github.com/yuxshao/ptcollab.git
+        tag: "v0.6.4.8"
+#        commit: "5feb27decb9e0600d1defdaf9dc1424c6fdcb7ad"
+    build-commands:
+      - mkdir /app/share/metainfo/
+      - cp io.github.yuxshao.ptcollab.metainfo.xml /app/share/metainfo/
```
[the remark this answers is not in the document]
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:27]: thanks

File: share/applications/io.github.yuxshao.ptcollab.desktop
Comment (germe-deb) [reply share/applications/io.github.yuxshao.ptcollab.desktop:1]: done

File: share/icons/hicolor/16x16/apps/io.github.yuxshao.ptcollab.png
Comment (germe-deb) [reply share/icons/hicolor/16x16/apps/io.github.yuxshao.ptcollab.png:1]: done

File: share/metainfo/io.github.yuxshao.ptcollab.metainfo.xml (hunk 4 above)
Comment (germe-deb) [reply share/metainfo/io.github.yuxshao.ptcollab.metainfo.xml]: done both

**Comment by hfiguiere** [comment]
bot, build io.github.yuxshao.ptcollab

**Comment by germe-deb** [comment]
Sorry! I needed to do that, otherwise it would fail

File: io.github.yuxshao.ptcollab.yml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  - --share=ipc
+  - --device=dri
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --socket=fallback-x11
+  - --filesystem=home
+modules:
+  - name: ptcollab
+    buildsystem: qmake
+    build-commands:
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:19]:
```suggestion
```

maybe that the failure you don't describe.

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:19]: Oh yeah, i thought i pushed that already

File: io.github.yuxshao.ptcollab.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  - name: ptcollab
+    buildsystem: qmake
+    sources:
+      - type: git
+        url: https://github.com/yuxshao/ptcollab.git
+        tag: "v0.6.4.8"
+#        commit: "5feb27decb9e0600d1defdaf9dc1424c6fdcb7ad"
+    build-commands:
+#      - mkdir /app/share/metainfo -p
+#      - cp io.github.yuxshao.ptcollab.metainfo.xml /app/share/metainfo/
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:26]: why did you ignore the suggestion I made for this?

File: io.github.yuxshao.ptcollab.yml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+modules:
+  - name: ptcollab
+    buildsystem: qmake
+    build-commands:
+    sources:
+      - type: git
+        url: https://github.com/yuxshao/ptcollab.git
+        tag: "v0.6.4.8"
+#        commit: "5feb27decb9e0600d1defdaf9dc1424c6fdcb7ad"
+    build-commands:
```
[the remark this answers is not in the document]
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:24]: yeah, I think it works now. the bad thing is that it seems that the icons doesn't work anymore

File: io.github.yuxshao.ptcollab.yml (hunk 10 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:26]: what

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:26]: sorry sorry sorry

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml:26]: i saw it

**Comment by germe-deb** [comment]
It seems to be totally finished right now, I think it's done

File: io.github.yuxshao.ptcollab.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml]: removed!!

File: io.github.yuxshao.ptcollab.yml (hunk 10 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml]: I removed that line now, it is running fine!

File: io.github.yuxshao.ptcollab.metainfo.xml
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: the developer name now is correct acording to himself!

File: io.github.yuxshao.ptcollab.metainfo.xml (hunk 7 above)
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml]: images are in github now

File: io.github.yuxshao.ptcollab.yml [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+sdk: org.kde.Sdk
+command: ptcollab
+rename-icon: ptcollab
+rename-desktop-file: ptcollab.desktop
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --device=dri
+  - --socket=pulseaudio
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:13]:
```suggestion
  - --socket=x11
```

since you removed wayland.

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.yml]: done

**Comment by germe-deb** [comment]
bot, build io.github.yuxshao.ptcollab

**Comment by germe-deb** [comment]
bot, build io.github.yuxshao.ptcollab

**Comment by germe-deb** [comment]
why this thread stopped pulling my commits? it says that I want to merge 22 commits and I have 25 in my repo

**Comment by germe-deb** [comment]
but that's already solved, just use the 3 commits I pushed

**Comment by germe-deb** [comment]
bot, build io.github.yuxshao.ptcollab

File: io.github.yuxshao.ptcollab.metainfo.xml
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.metainfo.xml:1]: this file is never installed. you removed the section that used to do it.

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: yeah, now i'm trying to figure out where to put it

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: I need help in this, I tried to use install and cp in build-commands, post-install and cleanup and I can't get it working, it crashes mid-compilation

File: io.github.yuxshao.ptcollab.metainfo.xml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+ 
+  <name>pxtone Collab</name>
+  <summary>Sample-based music editor where you can collaborate with friends!</summary>
+  <developer_name>pxtone collab</developer_name>
+  
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
+
+  <screenshots>
+    <screenshot type="default"><image>https://github.com/germe-deb/io.github.yuxshao.ptcollab/blob/58ce0534e5fb0d7e230eb196418c9c5ecf4a0208/screenshots/1stscreenshot.png</image></screenshot>
```
Comment (bbhtt) [inline io.github.yuxshao.ptcollab.metainfo.xml:13]:
```suggestion
    <screenshot type="default"><image>https://raw.githubusercontent.com/germe-deb/io.github.yuxshao.ptcollab/58ce0534e5fb0d7e230eb196418c9c5ecf4a0208/screenshots/1stscreenshot.png</image></screenshot>
```

File: io.github.yuxshao.ptcollab.metainfo.xml [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+    <screenshot><image>https://github.com/germe-deb/io.github.yuxshao.ptcollab/blob/58ce0534e5fb0d7e230eb196418c9c5ecf4a0208/screenshots/2ndscreenshot.png</image></screenshot>
```
Comment (bbhtt) [inline io.github.yuxshao.ptcollab.metainfo.xml:14]:
```suggestion
    <screenshot><image>https://raw.githubusercontent.com/germe-deb/io.github.yuxshao.ptcollab/58ce0534e5fb0d7e230eb196418c9c5ecf4a0208/screenshots/2ndscreenshot.png</image></screenshot>
```

File: metainfo.patch
Comment (bbhtt) [inline metainfo.patch:1]: Drop this

Comment (germe-deb) [reply metainfo.patch:1]: how do I solve this? I need that xml

File: io.github.yuxshao.ptcollab.metainfo.xml
Comment (bbhtt) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: Add this file as a source `type: file` `path: io.github.yuxshao.ptcollab.metainfo.xml` then use `post-install` to install it as `${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml`

File: metainfo.patch
Comment (bbhtt) [reply metainfo.patch:1]: See comment here https://github.com/flathub/flathub/pull/5007#discussion_r1509289471

File: io.github.yuxshao.ptcollab.metainfo.xml
Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: Ok!

Comment (germe-deb) [reply io.github.yuxshao.ptcollab.metainfo.xml:1]: I'm having trobule finding where to put post-install
everywhere I put it throws an error

File: io.github.yuxshao.ptcollab.yml [hunk 15]
```diff
@@ -19,5 +19,8 @@ modules:
       - type: git
         url: https://github.com/yuxshao/ptcollab.git
         tag: "v0.6.4.8"
-      - type: patch
-        path: metainfo.patch
+      - type: file
+        path: io.github.yuxshao.ptcollab.metainfo.xml
+    post-install:
+      - mkdir -p/app/share/metadata/
+      - cp -v io.github.yuxshao.ptcollab.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
```
Comment (hfiguiere) [inline io.github.yuxshao.ptcollab.yml:26]:
```suggestion
      - install -Dm644 io.github.yuxshao.ptcollab.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
```

I already suggested this in the past and was ignored. But your commands won't work either. A local build would have told you so.

**Comment by germe-deb** [comment]
still doesn't work, and I don't know why

**Comment by bbhtt** [comment]
bot, build io.github.yuxshao.ptcollab

**Comment by germe-deb** [comment]
testing time

**Comment by germe-deb** [comment]
It seems to work fine!

**Comment by germe-deb** [comment]
This is ready to go. some friends tested the build and works just like the other versions of the program.

**Comment by germe-deb** [comment]
bot, build io.github.yuxshao.ptcollab

**Comment by germe-deb** [comment]
It's ready to upload :)

**Pull request closed without merging**